Modules/posixmodule.c

```diff
@@ -5490,12 +5490,89 @@ os__path_normpath_impl(PyObject *module, PyObject *path)
         return NULL;
     }
     Py_ssize_t norm_len;
-    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, &norm_len);
+    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, 0, &norm_len);
     PyObject *result = PyUnicode_FromWideChar(norm_path, norm_len);
     PyMem_Free(buffer);
     return result;
 }
 
+#ifndef MS_WINDOWS
+
+/*[clinic input]
+os._path_abspath
+
+    path: unicode
+    /
+
+Make path absolute.
+[clinic start generated code]*/
+
+static PyObject *
+os__path_abspath_impl(PyObject *module, PyObject *path)
+/*[clinic end generated code: output=b58956d662b60be0 input=577ecb3473d22113]*/
+{
+    Py_ssize_t rel_path_len;
+    wchar_t *rel_path = PyUnicode_AsWideCharString(path, &rel_path_len);
+    if (!rel_path) {
+        return NULL;
+    }
+
+    if (rel_path[0] == '\0' || !wcscmp(rel_path, L".")) {
+        return posix_getcwd(0);
```

Comment:
It's better to ensure nothing is getting leaked if all cleanup is handled at a single exit point. Label the final section `exit`, where the working buffers are freed and `result` is returned. At the start of the function, declare `PyObject *result = NULL` and initialize the buffer pointers to `NULL`. Then change this to `result = posix_getcwd()` followed by `goto exit`.